cmd/gateway/hdfs/gateway-hdfs.go

```diff
@@ -507,12 +507,29 @@ func (n *hdfsObjects) GetObject(ctx context.Context, bucket, key string, startOf
 	return hdfsToObjectErr(ctx, err, bucket, key)
 }
 
+func (n *hdfsObjects) isObjectDir(bucket, object string) bool {
+	f, err := n.clnt.Open(minio.PathJoin(hdfsSeparator, bucket, object))
+	if err != nil {
```

Comment:
I really see it is better if we return an error with the bool, and check the error to return a correct {true,false} with a nil error.

The reason is that we don't know what is the consequence (disastrous?) if we return "false" when there is a network error for example.

Reply:
We don't need to be so conclusively sure.  We validate the bucket before already and it is highly unlikely that there will be a network issue as such.

Reply:
> We don't need to be so conclusively sure. We validate the bucket before already and it is highly unlikely that there will be a network issue as such.

Can we at least log this error when the error is unusual ? At least we can avoid rare issues that we don't understand in the future.

Reply:
This is done